crates/l2/contracts/src/l1/CommonBridge.sol

```diff
@@ -19,4 +29,26 @@ contract CommonBridge is ICommonBridge {
     receive() external payable {
         deposit(msg.sender);
     }
+
+    /// @inheritdoc ICommonBridge
+    function startWithdrawal(
+        WithdrawalTransaction[] calldata transactions
+    ) public onlyOwner {
+        for (uint256 i = 0; i < transactions.length; i++) {
+            pendingWithdrawals[transactions[i].l2TxHash] = WithdrawalData(
+                transactions[i].to,
+                transactions[i].amount
+            );
+        }
+    }
+
+    function finalizeWithdrawal(bytes32 l2TxHash) public {
+        require(
+            msg.sender == pendingWithdrawals[l2TxHash].to,
+            "CommonBridge: withdrawal not found"
+        );
+
+        payable(msg.sender).transfer(pendingWithdrawals[l2TxHash].amount);
```

Comment:
It's generally not a good idea to use the `transfer` or `send` methods; we should replace this with a `call` and add a reentrancy guard to this function